include/seqan3/std/span.hpp

```diff
@@ -0,0 +1,551 @@
+    }
+
+    constexpr reference operator()(index_type idx) const noexcept
+    {
+        static_assert(idx >= 0 && idx < get_size(), "span<T>() index out of bounds");
+        return data[idx];
+    }
+
+    constexpr pointer get_data()                         const noexcept { return data; }
+
+// [span.iter], span iterator support
+    constexpr iterator                 begin() const noexcept { return iterator(get_data()); }
+    constexpr iterator                   end() const noexcept { return iterator(get_data() + get_size()); }
+    constexpr const_iterator          cbegin() const noexcept { return const_iterator(get_data()); }
+    constexpr const_iterator            cend() const noexcept { return const_iterator(get_data() + get_size()); }
+    constexpr reverse_iterator        rbegin() const noexcept { return reverse_iterator(end()); }
+    constexpr reverse_iterator          rend() const noexcept { return reverse_iterator(begin()); }
+    constexpr const_reverse_iterator crbegin() const noexcept { return const_reverse_iterator(cend()); }
+    constexpr const_reverse_iterator   crend() const noexcept { return const_reverse_iterator(cbegin()); }
+
+    constexpr void swap(span &other) noexcept
+    {
+        pointer p = data;
+        data = other.data;
+        other.data = p;
+
+        index_type sz = size;
+        size = other.size;
+        other.size = sz;
+    }
+
+    span<const byte, dynamic_extent> as_bytes() const noexcept
+    { return {reinterpret_cast<const byte *>(get_data()), size_bytes()}; }
+
+    span<byte, dynamic_extent> as_writeable_bytes() const noexcept
+    { return {reinterpret_cast<byte *>(get_data()), size_bytes()}; }
+
+private:
+    pointer    data;
+    index_type size;
+};
+
+//  as_bytes & as_writeable_bytes
+template <class span_tp, ptrdiff_t span_extent>
+    auto as_bytes(span<span_tp, span_extent> s) noexcept
+    -> decltype(s.as_bytes())
+    { return s.as_bytes(); }
+
+template <class span_tp, ptrdiff_t span_extent>
+    auto as_writeable_bytes(span<span_tp, span_extent> s) noexcept
+    -> typename enable_if<!is_const_v<span_tp>, decltype(s.as_writeable_bytes())>::type
+    { return s.as_writeable_bytes(); }
+
+template <class span_tp, ptrdiff_t span_extent>
+    constexpr void swap(span<span_tp, span_extent> &lhs, span<span_tp, span_extent> &rhs) noexcept
+    { lhs.swap(rhs); }
+
+
+//  Deduction guides
+template<class span_tp, size_t span_sz>
+    span(span_tp (&)[span_sz]) -> span<span_tp, span_sz>;
+
+template<class span_tp, size_t span_sz>
+    span(array<span_tp, span_sz>&) -> span<span_tp, span_sz>;
+
+template<class span_tp, size_t span_sz>
+    span(const array<span_tp, span_sz>&) -> span<const span_tp, span_sz>;
+
+template<class container>
+    span(container&) -> span<typename container::value_type>;
+
+template<class container>
+    span(const container&) -> span<const typename container::value_type>;
+} // namespace std
+
+#endif // __cplusplus <= 201703L
```

Comment:
and close the no-documentation scope

```suggestion
#endif // __cplusplus <= 201703L
//!\endcond
```